ons_alpha/jinja2/templates/pages/topics/topic_page.html

```diff
@@ -42,6 +53,39 @@ <h2 class="ons-timeline__heading">
                         </div>
                     </section>
                 {% endfor %}
+
+                {% if page.related_by_topic %}
+                    <section id="related">
+                        <h2>{{ _("Related content") }}</h2>
+                        <p>{% trans topic=page.topic %}This sections surfaces content tagged with "{{ topic }}"{% endtrans %}</p>
+
+                        {% set rows=[] %}
+                        {% for type, items in page.related_by_topic.items() %}
+                            {% set items_list=[] %}
+                            {% for item in items %}
+                                {% do items_list.append({"url": pageurl(item), "text": item.full_title or item.title}) %}
+                            {% endfor %}
+                            {# fmt:off #}
+                            {%
+                                do rows.append({
+                                    "id": "related-content-" + type|slugify(),
+                                    "title": type,
+                                    "itemsList": items_list
+                                })
+                            %}
+                            {# fmt:on #}
+                        {% endfor %}
```

Comment:
Issue: I feel like this would be much easier to read / maintain (and probably faster) if it were in a method on the page itself, rather than in the template.

Reply:
Don't forget this is throwaway code. I agree, but acceptable for alpha.

Reply:
What Mebin said :)